packages/angular_devkit/build_angular/src/differential-loading/create-webpack-multi-config.ts

```diff
@@ -0,0 +1,185 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+
+//  This updates a webpack configuration which has been created by a
+//  webpack-based builder for differential loading. It duplicates an
+//  existing one and modifies e. g. the name of the bundles to generate
+//  ([name].modern.js instead of [name].js), the resolve array, and
+//  the Angular Compiler Plugin’s reference to the tsconfig.json to use.
+//
+//  see: https://docs.google.com/document/d/13k84oGwrEjwPyAiAjUgaaM7YHJrzYXz7Cbt6CwRp9N4/
+
+
+import { AngularCompilerPlugin } from '@ngtools/webpack';
+import * as webpack from 'webpack';
+
+const StatsPlugin = require('stats-webpack-plugin');
+
+
+export function createWebpackMultiConfig(config: webpack.Configuration): webpack.Configuration[] {
+
+  const legacyConfig: webpack.Configuration = {
+    ...config,
+    output: buildLegacyOutput(config),
+    plugins: buildLegacyPlugins(config),
+    resolve: buildLegacyResolve(config),
+  };
+
+  const modernConfig: webpack.Configuration = {
```

Comment:
With this approach you are missing some configuration for es2015 https://github.com/angular/angular-cli/blob/master/packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts#L236 and https://github.com/angular/angular-cli/blob/master/packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts#L216

Hence, I strongly recommend that we avoid doing such transformations and let the webpack config builder return the appropriate configs